lib/data/multipleBackendGateway.js

```diff
@@ -216,6 +218,90 @@ const multipleBackendGateway = {
             return callback(null, multBackendResp);
         });
     },
+
+    createMPU: (key, bucket, metaHeaders, websiteRedirectHeader,
+    location, log, cb) => {
+        const client = clients[location];
+        const awsKey = _createAwsKey(bucket, key, client.bucketMatch);
+        if (client.clientType === 'aws_s3') {
+            const params = { Bucket: client.awsBucketName, Key: awsKey,
+                WebsiteRedirectLocation: websiteRedirectHeader,
+                Metadata: metaHeaders };
+            return client.createMultipartUpload(params, (err, data) => {
+                if (err) {
+                    log.error('err from data backend',
+                    { error: err, dataStore: client.dataStoreName });
+                    return cb(errors.InternalError);
+                }
+                return cb(null, data);
+            });
+        }
+        const scalResponse = {};
+        scalResponse.UploadId = UUID.v4().replace(/-/g, '');
+        return cb(null, scalResponse);
+    },
+
+    uploadPart: (bucket, location, key, uploadId, partNumber, log, cb) => {
+        const client = clients[location];
+        if (client.clientType === 'aws_s3') {
+            const awsKey = _createAwsKey(bucket, key, client.bucketMatch);
+            const params = { Bucket: client.awsBucketName, Key: awsKey,
+                UploadId: uploadId, PartNumber: partNumber };
+            return client.uploadPart(params, (err, data) => {
+                if (err) {
+                    log.error('err from data backend',
+                    { error: err, dataStoreName: client.dataStoreName });
+                    return cb(errors.InternalError);
+                }
+                const dataRetrievalInfo = {
+                    key,
+                    dataStoreName: client.dataStoreName,
+                    dataStoreETag: data.ETag,
+                };
+                return cb(null, dataRetrievalInfo);
+            });
+        }
+        return cb();
+    },
+
+    completeMPU: (key, uploadId, location, jsonList, log, cb) => {
+        const client = clients[location];
```

Comment:
don't you still need to recreate the awskey?

why do you use the bucket as an argument in put part but pull the bucket from the client here?  shouldn't we always pull the bucket from the client info in case there is not a bucket match?